Population.cpp

```diff
@@ -23,3 +26,111 @@ const Individual &Population::operator[](int index) const {
     return population_vector_[index];
 }
 
+Individual Population::generate_individual() {
+    /* This method is used to create a parent for the next generation of offsprings
+     *  or an individual for the next generation using the wheel algorithm
+     *  which operates with the partial sums of the probabilities
+     *  of the individuals from the populations
+     */
+
+    std::vector<double> individuals_probabilities;
+    std::vector<double> cumulative_ascending_probabilities;
+
+    calculate_individuals_probabilities(individuals_probabilities);
+    calculate_cumulative_ascending_probabilities(individuals_probabilities, cumulative_ascending_probabilities);
```

Comment:
These functions shouldn't be called for each individual, but rather in generate_offsprings()